File: src/csharp/SlimShader/Chunks/Shex/Tokens/NameToken.cs
using SlimShader.Util;

namespace SlimShader.Chunks.Shex.Tokens
{
	/// <summary>
	/// Name Token (NameToken) (used in declaration statements)
	///
	/// [15:00] D3D10_SB_NAME enumeration
	/// [31:16] Reserved, 0
	/// </summary>
	public static class NameToken
	{
		public static SystemValueName Parse(BytecodeReader reader)
		{
			uint token = reader.ReadUInt32();
			return token.DecodeValue<SystemValueName>(0, 15);
		}
	}
}

File: src/cpp/Source/Chunks/Shex/Tokens/NameToken.cpp
#include "PCH.h"
#include "NameToken.h"

#include "Decoder.h"

using namespace SlimShader;

SystemValueName NameToken::Parse(BytecodeReader& reader)
{
	auto token = reader.ReadUInt32();
	return DecodeValue<SystemValueName>(token, 0, 15);
}